src/libz.jl

```diff
@@ -99,6 +118,11 @@ function inflate_end!(zstream::ZStream)
     return ccall((:inflateEnd, libz), Cint, (Ref{ZStream},), zstream)
 end
 
+function decompress_finalizer!(zstream::ZStream)
+    inflate_end!(zstream)
```

Comment:
What happens if this gets called twice?

Reply:
Nothing.

`InflateEnd` starts by checking if `zstream.state` is null and returns `Z_STREAM_ERROR` if it is https://github.com/madler/zlib/blob/8a844d434f0eef87d972ae6406b00968f7c52944/inflate.c#L1268-L1269

`InflateEnd` also ends by setting `zstream.state` to null. https://github.com/madler/zlib/blob/8a844d434f0eef87d972ae6406b00968f7c52944/inflate.c#L1273

The same happens in `DeflateEnd` https://github.com/madler/zlib/blob/8a844d434f0eef87d972ae6406b00968f7c52944/deflate.c#L1266-L1283

Also, since the `ZStream` constructor initializes `.state` to `C_NULL`, the finalizer can run on uninitialized streams without doing anything.